t/501-hooks-yaml-metadata.t

```diff
@@ -0,0 +1,52 @@
+use strict;
+use warnings;
+use utf8;
+
+use Markdown::Perl 'convert', 'set_hooks';
+use Test2::V0;
+
+my $p = Markdown::Perl->new();
+my $page = <<EOF;
+---
+name: Mark is down
+draft: false
+number: 42
+---
+# Mark is down!
+
+I repeat: "Mark is down!"
+EOF
+
+my $invalid_page = <<EOF;
+---
+name: Mark is down
+  draft: false
+	number: 42
+---
+# Mark is down!
+
+I repeat: "Mark is down!"
+EOF
+
+# Test 1: Check if we can get a string value
+{
+  sub hook_is_name_mark {
+    my $x = shift;
+    ok(exists($x->[0]->{name}) && $x->[0]->{name} eq 'Mark is down', "key 'name' was retrieved and validated as being 'Mark is down'");
```

Comment:
Given that the initial table can only contain a single YAML document, maybe we can simplify things for the caller by passing it directly the reference to that first document, rather than the whole YAML::Tiny object.

Reply:
Yup, sounds like a good idea. Then we also hide the YAML::Tiny implementation detail from users.